core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java

```diff
@@ -42,24 +44,26 @@ default List<HostAddress> getAddresses()
         return List.of();
     }
 
-    @JsonIgnore // ConnectorSplit is json-serializable, but we don't want to repeat information in that field
+    @JsonIgnore // TODO remove after https://github.com/airlift/airlift/pull/1141
```

Comment:
Why this TODO?
Is there an assumption that implementation will always be a record? how would that be enforced?

Reply:
This needs to be explicitly ignored is for records only, AFAICT.
If the implementation is not a record, getter-like methods are not treated as fields to be serialized to JSON.